src/Core/Form/IdentifiableObject/DataProvider/CategoryFormDataProvider.php

```diff
@@ -82,6 +100,53 @@ public function getData($categoryId)
         /** @var EditableCategory $editableCategory */
         $editableCategory = $this->queryBus->handle(new GetCategoryForEditing($categoryId));
 
+        $coverImages = $thumbnailImages = $menuThumbnailImagesEdited = null;
+        $disableMenuThumbnailsUpload = false;
+        if ($categoryId) {
```

Comment:
You're not supposed to check this, `getData` is only used bor edition and the ID MUST be present